modules/service/src/main/scala/lucuma/odb/graphql/table/GmosDynamicTables.scala

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+package table
+
+import lucuma.odb.util.Codecs.*
+import lucuma.odb.util.GmosCodecs.*
+import skunk.codec.text.varchar
+
+trait GmosDynamicTables[F[_]] extends BaseMapping[F] {
+
+  class GmosDynamicTable[G, L, U](
+    name:    String,
+    grating: skunk.Codec[G],
+    filter:  skunk.Codec[L],
+    fpu:     skunk.Codec[U]
+  ) extends TableDef(name) {
+    val Id: ColumnRef            = col("c_step_id",           step_id)
+    val Instrument: ColumnRef    = col("c_instrument",        instrument)
+
+    val ExposureTime: ColumnRef  = col("c_exposure_time",     time_span)
+
+    object CcdMode {
+      val Xbin: ColumnRef        = col("c_xbin",              gmos_x_binning)
+      val Ybin: ColumnRef        = col("c_ybin",              gmos_y_binning)
+      val AmpCount: ColumnRef    = col("c_amp_count",         gmos_amp_count)
+      val AmpGain: ColumnRef     = col("c_amp_gain",          gmos_amp_gain)
+      val AmpReadMode: ColumnRef = col("c_amp_read_mode",     gmos_amp_read_mode)
+    }
+
+    val Dtax: ColumnRef          = col("c_dtax",              gmos_dtax)
+    val Roi: ColumnRef           = col("c_roi",               gmos_roi)
+
+    // Grating (if any)
+    object Grating {
+      val Disperser: ColumnRef  = col("c_grating_disperser",  grating.embedded)
+      val Order: ColumnRef      = col("c_grating_order",      gmos_grating_order.embedded)
+      val Wavelength: ColumnRef = col("c_grating_wavelength", wavelength_pm.embedded)
+    }
```

Comment:
I flipped this from `opt` to `embedded` when I got an incomprehensible error. Yet the `Fpu` definition below seems to need `opt` and not `embedded`.  It seems to work but a critical, informed look would be appropriate.

Reply:
If it works I would just merge and maybe open an issue to review it.